identity-apps-core/apps/recovery-portal/src/main/webapp/self-registration-with-verification.jsp

```diff
@@ -105,6 +105,12 @@
         tenantDomain = srtenantDomain;
     }
 
+    if (username.contains(IdentityManagementEndpointConstants.TENANT_DOMAIN_SEPARATOR) && tenantDomain != null) {
+        if (username.split(IdentityManagementEndpointConstants.TENANT_DOMAIN_SEPARATOR)[1] != tenantDomain) {
+            username = username + IdentityManagementEndpointConstants.TENANT_DOMAIN_SEPARATOR + tenantDomain;
```

Comment:
What happens if the tenant domain and the domain name of the email is same. eg: tenant domain abc.com and the email is user@abc.com.

Reply:
Fixes with a20562c8c8c55669fd36d827e4d16e6863b0c746